### PR TITLE
Implement secure logging and inspection

### PR DESCRIPTION
**Pull request by doomspork**

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use Mix.Config
+
+patterns = [
+  # Credit Cards
+  ~r/#?(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})/,
```
Comment (doomspork) [inline config/config.exs:5]: These won't be included with the package, maybe we should create a more complete list and ensure they are included?

File: lib/secure_log_formatter.ex [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import Kernel, except: [inspect: 1]
+
+defmodule SecureLogFormatter do
+  @moduledoc """
+  Documentation for SecureLogFormatter.
+  """
+
+  alias Logger.Formatter
+
+  @default_fields ["password"]
```
Comment (doomspork) [inline lib/secure_log_formatter.ex]: What other fields do we want to redact by default?

File: lib/secure_log_formatter.ex [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  @format_config config
+    |> Keyword.get(:format, @default_pattern)
+    |> Formatter.compile
+
+
+  def format(level, msg, ts, md),
+    do: Formatter.format(@format_config, level, sanitize(msg), ts, md)
+
+  @doc """
+  Securely inspect a Map
```
Comment (doomspork) [inline lib/secure_log_formatter.ex]: Update the documentation for this function.

File: lib/secure_log_formatter.ex [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  # Examples
+
+      iex> SecureLogFormatter.inspect(%{username: "doomspork", password: "abc123"})
```
Comment (doomspork) [inline lib/secure_log_formatter.ex]: Add additional doctests.

File: config/config.exs (hunk 1 above)
Comment (joshleecreates) [reply config/config.exs:5]: I could get behind that. I'd also want to see key/value pairs so specific patterns can easily be turned on or off depending on the use case

File: mix.exs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+defmodule SecureLogFormatter.Mixfile do
+  use Mix.Project
+
+  @version "0.1.0"
+
+  def project do
+    [app: :secure_log_formatter,
+     build_embedded: Mix.env == :prod,
+     deps: deps(),
+     description: "Emque compatible producers and consumers for Elixir.",
```
Comment (joshleecreates) [inline mix.exs]: This seems not right

File: lib/secure_log_formatter.ex [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      iex> SecureLogFormatter.inspect(%{user: "username", password: "abc123"})
+      ~s(%{password: \"[REDACTED]\", user: \"username\"})
+
+      iex> SecureLogFormatter.inspect([access_token: "secret"])
+      ~s([access_token: \"[REDACTED]\"])
+
+      iex> SecureLogFormatter.inspect("Customer CC 4111111111111111")
+      ~s("Customer CC [REDACTED]")
+  """
+  def inspect(message) do
```
Comment (joshleecreates) [inline lib/secure_log_formatter.ex:48]: Can you tell me how this function gets called? Would one always call this directly or would it be possible to override the behavior of kernel inspect?

File: mix.exs (hunk 5 above)
Comment (doomspork) [reply mix.exs]: Copy pasta!  🍝

File: config/config.exs (hunk 1 above)
Comment (doomspork) [reply config/config.exs:5]: In those situations I'd rather see the developer to define their own patterns than us try to get too crazy.  I'm trying to think of a few sane defaults out-of-the-box for folks who have no interest in configuration but want to make sure some of the more obvious things are addressed.

File: lib/secure_log_formatter.ex (hunk 6 above)
Comment (doomspork) [reply lib/secure_log_formatter.ex:48]: @joshleecreates check the tests for how I handled this.  For now you need to do something like:


```elixir
import Kernel, except: [inspect: 1]
import SecureLogFormatter, only: [inspect: 1]
```

File: config/config.exs (hunk 1 above)
Comment (joshleecreates) [reply config/config.exs:5]: 👍

File: lib/secure_log_formatter.ex (hunk 6 above)
Comment (joshleecreates) [reply lib/secure_log_formatter.ex:48]: 👍

File: mix.exs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+defmodule SecureLogFormatter.Mixfile do
+  use Mix.Project
+
+  @version "0.1.0"
```
Comment (doomspork) [inline mix.exs]: I'm thinking I want to cut this as `1.0.0`.

**Pull request merged**